### PR TITLE
[8.0][product_supplierinfo_discount] Include discount in POL created from Procurement

### PR DESCRIPTION
**Pull request by andrius-preimantas**
Include discount when Purchase Order Line is created from Procurement.

### CODE REVIEW

File: product_supplierinfo_discount/models/procurement_order.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/procurement_order.py]: Use new short header styling: https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py

File: product_supplierinfo_discount/models/purchase_order.py [hunk 2]
```diff
@@ -57,11 +49,33 @@ def onchange_product_id(
...
+            date_order=date_order, fiscal_position_id=fiscal_position_id,
+            date_planned=date_planned, name=name, price_unit=price_unit,
+            state=state)
+        if not product_id:
+            return res
+
+        # Look for a possible discount
+        discount = self._get_product_discount(product_id, qty, partner_id)
+        if discount is not None:
+            if 'value' not in res:
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/purchase_order.py]: You can make:

```
val = res.setdefault('value', {})
val['discount'] = discount
```

File: product_supplierinfo_discount/models/purchase_order.py [hunk 3]
```diff
@@ -25,23 +25,15 @@
...
-            return res
+    @api.model
+    def _get_product_discount(self, product_id, qty, partner_id):
+        """Return discount for provided qty and supplier or None otherwise."""
         supplierinfo_obj = self.env['product.supplierinfo']
         product_obj = self.env['product.product']
         product_uom_obj = self.env['product.uom']
         pl_pinfo_obj = self.env['pricelist.partnerinfo']
-        # Look for a possible discount
+
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/purchase_order.py]: Remove these empty lines in the middle of the method.

**Comment by legalsylvain** [comment]
:+1: LGTM. Code review. Thanks.

could you just take make @pedrobaeza suggestions ? 

kind regards.

**Comment by andrius-preimantas** [comment]
Changes added based on @pedrobaeza comments.

Cheers

**Comment by pedrobaeza** [comment]
Thanks :+1:

**Comment by legalsylvain** [comment]
thanks !

**Pull request merged**